<launch>
<!-- Launch file for optimized point cloud visualization on edge devices -->

    <!-- Include the main mapping launch file -->
    <include file="$(find fast_livo)/launch/mapping_mid360.launch">
        <arg name="rviz" value="false" />
    </include>

    <!-- PointCloud Optimizer Node -->
    <node pkg="fast_livo" type="pointcloud_optimizer" name="pointcloud_optimizer" output="screen">
        <param name="voxel_size" value="0.2" />
        <param name="publish_rate" value="5" />
        <param name="enable_compression" value="true" />
    </node>

    <!-- Lightweight RViz configuration -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find fast_livo)/rviz_cfg/optimized_rviz.rviz" output="screen">
        <param name="synchronization" value="false" />
        <param name="queue_size" value="1" />
    </node>

</launch>
